src/storage/src/source/persist_source.rs

```diff
@@ -132,91 +128,40 @@ where
+                            Some(ts) => {
+                                current_ts = ts;
+                                cap.downgrade(&ts);
                             }
+                            None => return SourceStatus::Done,
                         },
-                        Some(Err::<_, anyhow::Error>(e)) => {
+                        Some(Ok(ListenEvent::Updates(mut updates))) => {
                             let cap = cap.delayed(&current_ts);
                             let mut session = output.session(&cap);
-                            session.give(Err((format!("{}", e), current_ts, 1)));
+                            session.give_vec(&mut updates);
                         }
-                        None => {
-                            unreachable!("We poll from persist continuously, the Stream should therefore never be exhausted.")
+                        Some(Err::<_, ExternalError>(e)) => {
+                            // TODO(petrosagg): error handling
+                            panic!("unexpected error from persist {e}");
                         }
-                    }
-
-                    if timer.elapsed() > YIELD_INTERVAL {
-                        let _ = activator.activate();
-                        return SourceStatus::Alive;
+                        None => return SourceStatus::Done,
                     }
                 }
 
                 SourceStatus::Alive
             }
-        },
-    );
-
-    let (ok_stream, persist_err_stream) = timely_stream.ok_err(|x| x);
-    let persist_err_stream = persist_err_stream.map(move |(err, ts, diff)| {
-        (
-            DataflowError::from(SourceError::new(
-                source_id,
-                SourceErrorDetails::Persistence(err),
-            )),
-            ts,
-            diff,
-        )
-    });
+        });
 
-    let mut row = Row::default();
-    let (ok_stream, err_stream) = ok_stream.ok_err(move |((key, value), ts, diff)| {
-        let mut row_packer = row.packer();
-
-        let key = match key {
-            Ok(key) => key,
-            Err(e) => return Err((DataflowError::from(DecodeError::Text(e)), ts, diff)),
-        };
-        let value = match value {
-            Ok(value) => value,
-            Err(e) => return Err((DataflowError::from(DecodeError::Text(e)), ts, diff)),
-        };
-
-        let unpacked_key = key.unpack();
-        let unpacked_value = value.unpack();
-        row_packer.extend(unpacked_key);
-        row_packer.extend(unpacked_value);
-
-        // TODO(aljoscha): Metrics about how many messages have been ingested from persist/STORAGE?
-        // metrics.messages_ingested.inc();
-        Ok((row.clone(), ts, diff))
+    let (ok_stream, err_stream) = timely_stream.ok_err(|x| match x {
+        ((Ok(()), Ok(SourceData(Ok(row)))), ts, diff) => Ok((row, ts, diff)),
+        ((Ok(()), Ok(SourceData(Err(err)))), ts, diff) => Err((err, ts, diff)),
+        // TODO(petrosagg): error handling
```

Comment:
nothing to do here but panic. either someone messed up a Codec impl or someone messed up a code rollout